### PR TITLE
fix pk constraint exc in IntegrationTest

### PR DESCRIPTION
**Pull request by patriknw**
References #534

### CODE REVIEW

File: docs-source/docs/modules/microservices-tutorial/examples/04-shopping-cart-service-java/src/test/java/shopping/cart/ItemPopularityIntegrationTest.java [hunk 1]
```diff
@@ -104,6 +104,7 @@ public void consumeCartEventsAndUpdatePopularityCount() throws Exception {
     assertEquals(4L, summary3.items.get(item2).intValue());
 
     probe.awaitAssert(
+        Duration.ofSeconds(10),
```
Comment (patriknw) [inline docs-source/docs/modules/microservices-tutorial/examples/04-shopping-cart-service-java/src/test/java/shopping/cart/ItemPopularityIntegrationTest.java:107]: What I think happens is that the test is adding item2 to two different carts and two different projects process those events at the same time, resulting in pk violation. It's supposed to retry, but I think it need some more time for that.

I will update all places if this works.

File: docs-source/docs/modules/microservices-tutorial/examples/04-shopping-cart-service-java/src/test/resources/logback-test.xml [hunk 2]
```diff
@@ -7,7 +7,7 @@
         </encoder>
     </appender>
 
-    <root level="INFO">
+    <root level="DEBUG">
```
Comment (patriknw) [inline docs-source/docs/modules/microservices-tutorial/examples/04-shopping-cart-service-java/src/test/resources/logback-test.xml:10]: will revert but wanted to see more if it still fails

**Comment by ignasi35** [comment]
Partially related to this, I think optimistic locking is not working as we expected. I'll try to fix it in #559.

File: docs-source/docs/modules/microservices-tutorial/examples/04-shopping-cart-service-java/src/test/java/shopping/cart/ItemPopularityIntegrationTest.java (hunk 1 above)
Comment (ignasi35) [reply docs-source/docs/modules/microservices-tutorial/examples/04-shopping-cart-service-java/src/test/java/shopping/cart/ItemPopularityIntegrationTest.java:107]: I did a bit of extra investigating on https://github.com/akka/akka-platform-guide/pull/559 and I think this PR fixes the flakiness.

**Review by ignasi35 — APPROVED** [review]

**Pull request merged**